scripts/linkbot/monitor-raspbian-os.js

```diff
@@ -1,67 +1,18 @@
-const jsdom = require("jsdom");
-const { JSDOM } = jsdom;
-const fs = require("fs");
-const { updateURLLink } = require("./common");
+const fs = require('fs');
+const { getLinksByMatch } = require('./common');
 
-const target = "RASPBIAN_URL";
+const reg = 'raspios_lite_armhf-(?<date>[0-9]{4}-[0-9]{2}-[0-9]{1,2})/.*';
+const target = 'RASPBIAN_URL';
 
-// Read the input file, and parse the variable input
-try {
-  const data = fs
-    .readFileSync("../../.gitlab-ci.yml", "utf8")
-    .split("\n")
-    .filter((line) => line.match(`${target}: .*`));
-  var line = data[0];
-  console.log(line);
-  var reg =
-    "raspios_lite_armhf-(?<date>[0-9]{4}-[0-9]{2}-[0-9]{1,2})/(?<updated>[0-9]{4}-[0-9]{2}-[0-9]{1,2}).*$";
-  var m = line.match(
-    ".*: \"(?<url>[a-zA-Z-://._]*)(?<imageName>raspios_lite_armhf-[0-9]{4}-[0-9]{2}-[0-9]{1,2})/(?<updated>[0-9]{4}-[0-9]{2}-[0-9]{1,2}).*\"$"
-  );
-  console.log(m);
-  var url = m.groups.url;
-  var imageName = m.groups.imageName;
-  var updated = m.groups.updated;
-} catch (err) {
-  console.error(err);
-  process.exit(1);
-}
-
-
-function getNewRaspbian(target, url, separator) {
-    console.log("getNewRaspbian");
-    console.log(`${url}`);
-    return JSDOM.fromURL(url, {}).then((dom) => {
-        var document = dom.window.document;
-        var refs = document.getElementsByTagName("a");
-        var test = Array.from(refs)
-            .filter((ref) => ref.textContent.match("raspios-.*-lite.*\.img\.(zip|xz)$"))
-            .reduce((acc, element) => {
-                acc.push(element.textContent.match("raspios-.*-lite.*\.img\.(zip|xz)$").input);
-                return acc;
-            }, [])[0];
-        console.log(`Test var: ${test}`);
-        const imageName = test.substring(0, test.lastIndexOf('.img'));
-        if (separator) {
-            return {
-                replacementLine: `${target}${separator} \"${url}/${test}\"`,
-                imageName: imageName,
-            };
-        }
-        return { replacementLine: `${target}=\"${url}/${test}\"`, imageName: imageName };
-    });
-}
-
-
-function updateGitlabCILink(newLine, newURL, newName) {
+const updateGitlabCILink = ({ name: targetName, url: targetUrl }, newName, newUrl) => {
   try {
     const data = fs
-      .readFileSync("../../.gitlab-ci.yml", "utf8")
+      .readFileSync('../../.gitlab-ci.yml', 'utf8')
```

Comment:
I guess, with a little bit of sweat, we could unify this with the regular `run.sh` updater?

Reply:
hmm I would imagine a useable approach would be to collect all the variables in a `.env` file and use this from both locations when needed... OTOH it might be easier to just make the read/write files part more reusable...

Reply:
Hmm, could be an idea